src/taskpane/server.js

```diff
@@ -1637,17 +1789,23 @@ function setFeedbackBgColor(cell, error, field, fieldTypeEnums, artifact, colors
 // @param: field - the field specific to the cell
 // @param: fieldTypeEnums - enum information about field types
 // @param: message - relevant error message from the entry for this row
-function setFeedbackNote(cell, error, field, fieldTypeEnums, message) {
+// @param: value - original ID of the artifact to be kept in case of an error
+function setFeedbackNote(cell, error, field, fieldTypeEnums, message, value, isUpdate) {
+  console.log('field:');
+  console.dir(field);
   // handle entries with errors - add error notes into ID field
   if (error && field.type == fieldTypeEnums.id) {
-    return message;
+    if (value == "-1" && isUpdate && field.field != "TestCaseId" && field.field != "TestStepId") {

```

Comment:
this magic number comes from Spira I think? Can you please add a comment explaining where -1 comes from and why it is being handled in this way?
Also avoid using hard coded field names here - I don't think POSTS handle test cases with steps in this way - eg with code like "if (artifact.hasSubType && !entry.isSubType)"

Reply:
Comments added and enumeration implemented to make the code easier to understand and more generic